### PR TITLE
VIMC-3571: Add function to run orderly report as RStudio job

### PR DESCRIPTION
**Pull request by r-ash**
The testing here is a bit fiddly

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -6,3 +6,7 @@ get_colours <- function() {
     white = "#FFFFFF"
   )
 }
+
+system_file <- function(...) {
```
Comment (richfitz) [inline R/utils.R]: can you make this orderly_rstudio_file?

File: inst/scripts/orderly_run.R [hunk 2]
```diff
@@ -0,0 +1,12 @@
+## Prepare params
+args <- readRDS(Sys.getenv("ORDERLY_RUN_ARGS_PATH"))
```
Comment (richfitz) [inline inst/scripts/orderly_run.R:8]: I think a short comment explaining that this is the easiest way of parametrising a job might help us in future

File: inst/scripts/orderly_run.R [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+orderly:::orderly_run_internal(name = args[["name"]],
+                               parameters = args[["parameters"]],
+                               envir = args[["envir"]], root = args[["root"]],
```
Comment (richfitz) [inline inst/scripts/orderly_run.R]: might as well do one per line consistently here (`tags` below too)

File: tests/testthat/helper-orderly.rstudio.R [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+append_lines <- function(text, filename) {
+  prev <- readLines(filename)
+  writeLines(c(prev, text), filename)
+}
+
+## Taken from rstudiapi test code see
+## https://github.com/rstudio/rstudioapi/blob/master/tests/testthat/helper-rstudio.R
+skip_if_not_rstudio <- function(version = NULL) {
+  available <- rstudioapi::isAvailable(version)
+  message <- if (is.null(version))
```
Comment (richfitz) [inline tests/testthat/helper-orderly.rstudio.R]: could you use braces on the 1-line if/elses please

File: R/orderly_run_session.R [hunk 5]
```diff
@@ -0,0 +1,26 @@
+orderly_run_session <- function(name = NULL, parameters = NULL, envir = NULL,
```
Comment (richfitz) [inline R/orderly_run_session.R:14]: How do you run this? Does it need exporting?

**Review by richfitz — CHANGES_REQUESTED** [review]

File: R/orderly_run_session.R (hunk 5 above)
Comment (r-ash) [reply R/orderly_run_session.R:14]: Oops, of course!

File: inst/scripts/orderly_run.R (hunk 2 above)
Comment (r-ash) [reply inst/scripts/orderly_run.R:8]: Thanks, added some notes to that.

File: .github/workflows/R-CMD-check.yaml [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+          key: ${{ runner.os }}-${{ hashFiles('.github/R-version') }}-1-${{ hashFiles('.github/depends.Rds') }}
+          restore-keys: ${{ runner.os }}-${{ hashFiles('.github/R-version') }}-1-
+
+      - name: Install system dependencies
+        if: runner.os == 'Linux'
+        run: |
+          while read -r cmd
+          do
+            eval sudo $cmd
+          done < <(Rscript -e 'writeLines(remotes::system_requirements("ubuntu", "20.04"))')
```
Comment (richfitz) [inline .github/workflows/R-CMD-check.yaml:64]: I had a fight with this on another project. You need to add

```
sudo apt-get update && sudo apt-get install -y libcurl4-openssl-dev
```

Look for the gert failure in the runner log. You might need to add libgit2-dev too. I'm surprised the auto thing is not doing this properly for us

**Review by richfitz — CHANGES_REQUESTED** [review]
Looking good. There's a hint about fixing the gha failure, and worth bumping the version too

**Review by richfitz — COMMENTED** [review]
I created a demo orderly project and ran 
```r
orderly.rstudio::orderly_run_session("minimal")
```

but got error
```
Error in orderly_locate_config() : 
  Reached root without finding 'orderly_config.yml'
Calls: sourceWithProgress ... initialize -> orderly_config -> orderly_locate_config
Execution halted
```

any ideas?

**Comment by r-ash** [comment]
> I created a demo orderly project and ran
> 
> ```r
> orderly.rstudio::orderly_run_session("minimal")
> ```
> 
> but got error
> 
> ```
> Error in orderly_locate_config() : 
>   Reached root without finding 'orderly_config.yml'
> Calls: sourceWithProgress ... initialize -> orderly_config -> orderly_locate_config
> Execution halted
> ```
> 
> any ideas?

This was because I wasn't setting the working directory in the RStudio job. I've added a test which recreates this and fixed now so it should work as expected - can you try again?

**Review by richfitz — APPROVED** [review]

**Pull request merged**